src/Tools/dotnet-monitor/Egress/EgressArtifactSettings.cs

```diff
@@ -21,6 +21,12 @@ internal sealed class EgressArtifactSettings
         public Dictionary<string, string> Metadata { get; }
             = new Dictionary<string, string>(StringComparer.Ordinal);
 
+        /// <summary>
+        /// Custom metadata of the blob to be created.
+        /// </summary>
+        public Dictionary<string, string> CustomMetadata { get; }
+            = new Dictionary<string, string>(StringComparer.Ordinal);
```

Comment:
On Linux, environment variable names are case sensitive, but on Windows and MacOS they are not. So `HOSTNAME` and `hostname` are the same variable on Windows and MacOS. We might need to merge case-insensitive matching variables on Windows and MacOS. But the fact that you are using `StringComparer.Ordinal` here I think is correct.

Reply:
Ah, yes, you are correct. I'm reading this backwards. The mapping is from metadata name to environment variable name, whereas I was reading as from environment variable name to metadata name. So no issue here.